In [14]:
%matplotlib inline

from path import Path

import math
import numpy as np
import pandas as pd
import matplotlib.pylab as plt
import statsmodels.formula.api as sm
from statsmodels.tsa import tsatools, stattools
from statsmodels.tsa.arima_model import ARIMA
from statsmodels.graphics import tsaplots

In [15]:
walmart = pd.read_csv('WalMartStock.csv')

In [16]:
walmart.head()

Date  Close
0  5-Feb-01  53.84
1  6-Feb-01  53.20
2  7-Feb-01  54.66
3  8-Feb-01  52.30
4  9-Feb-01  50.40

In [17]:
walmart.tail()

Date  Close
243  29-Jan-02  57.91
244  30-Jan-02  59.75
245  31-Jan-02  59.98
246   1-Feb-02  59.26
247   4-Feb-02  58.90

In [18]:
len(walmart)

248

In [19]:
# convert date to first of each month
walmart['Date'] = pd.to_datetime(walmart.Date, format='%d-%b-%y').dt.to_period('M')
walmart_ts = pd.Series(walmart.Close.values, index=walmart.Date, name='walmart')
walmart_arima = ARIMA(walmart_ts, order=(1, 0, 0)).fit(disp=0)
print(pd.DataFrame({'coef': walmart_arima.params, 'std err': walmart_arima.bse}))

                    coef   std err
const          52.951056  1.328334
ar.L1.walmart   0.955839  0.018709


The AR(1) slope coefficient is relevant for testing whether this walmart stock is a random walk

Here the slope coefficient is 0.955, with a standard error of 0.019, indicating that this is a random walk. Forecasting this series using any of the methods described earlier (aside from the naive forecase) is therefore futile

it is impossible to obtain forecasts that are more accurate than naive forecasts for the series